server/src/utils/getConfig.ts

```diff
@@ -12,36 +12,84 @@ import {
+    microsoftBookingsUrl: process.env[VV_MICROSOFT_BOOKINGS_URL_ENV_NAME] ?? defaultConfig.microsoftBookingsUrl,
     chatEnabled:
       typeof process.env[VV_CHAT_ENABLED_ENV_NAME] === 'string'
         ? process.env[VV_CHAT_ENABLED_ENV_NAME]?.toLowerCase() === 'true'
-        : DefaultConfig.chatEnabled,
+        : defaultConfig.chatEnabled,
     screenShareEnabled:
       typeof process.env[VV_SCREENSHARE_ENABLED_ENV_NAME] === 'string'
         ? process.env[VV_SCREENSHARE_ENABLED_ENV_NAME]?.toLowerCase() === 'true'
-        : DefaultConfig.screenShareEnabled,
-    companyName: process.env[VV_COMPANY_NAME_ENV_NAME] ?? DefaultConfig.companyName,
-    colorPalette: process.env[VV_COLOR_PALETTE_ENV_NAME] ?? DefaultConfig.colorPalette,
-    waitingTitle: process.env[VV_WAITING_TITLE_ENV_NAME] ?? DefaultConfig.waitingTitle,
-    waitingSubtitle: process.env[VV_WAITING_SUBTITLE_ENV_NAME] ?? DefaultConfig.waitingSubtitle,
-    logoUrl: process.env[VV_LOGO_URL_ENV_NAME] ?? DefaultConfig.logoUrl
-  };
+        : defaultConfig.screenShareEnabled,
+    companyName: process.env[VV_COMPANY_NAME_ENV_NAME] ?? defaultConfig.companyName,
+    colorPalette: process.env[VV_COLOR_PALETTE_ENV_NAME] ?? defaultConfig.colorPalette,
+    waitingTitle: process.env[VV_WAITING_TITLE_ENV_NAME] ?? defaultConfig.waitingTitle,
+    waitingSubtitle: process.env[VV_WAITING_SUBTITLE_ENV_NAME] ?? defaultConfig.waitingSubtitle,
+    logoUrl: process.env[VV_LOGO_URL_ENV_NAME] ?? defaultConfig.logoUrl
+  } as ServerConfigModel;
+
+  config.postCall = getPostCallConfig(defaultConfig);
+
+  return config;
+};
+
+const isValidPostCallSurveyType = (postcallSurveyType: string): postcallSurveyType is PostCallSurveyType => {
+  return ['msforms', 'thirdparty'].indexOf(postcallSurveyType) !== -1;
+};
+
+const getPostCallConfig = (defaultConfig: ServerConfigModel): PostCallConfig | undefined => {
+  let postcallConfig: PostCallConfig | undefined;
+  //Setting values for postcallconfig from defaultconfig values first (if valid)
+  if (
+    defaultConfig.postCall?.survey?.type &&
+    isValidPostCallSurveyType(defaultConfig.postCall?.survey?.type) &&
+    defaultConfig.postCall?.survey?.options?.surveyUrl
+  ) {
+    postcallConfig = {
+      survey: {
+        type: defaultConfig.postCall?.survey?.type,
+        options: { surveyUrl: defaultConfig.postCall?.survey?.options?.surveyUrl }
+      }
+    };
+  }
+
+  try {
+    const postcallSurveyType = process.env[VV_POSTCALL_SURVEY_TYPE_ENV_NAME];
+    const postcallSurveyUrl = process.env[VV_POSTCALL_SURVEY_OPTIONS_SURVEYURL_ENV_NAME];
+
+    if (
+      postcallSurveyType &&
+      (postcallSurveyType === 'msforms' || postcallSurveyType === 'thirdparty') &&
+      postcallSurveyUrl
+    ) {
+      postcallConfig = {
+        survey: { type: postcallSurveyType, options: { surveyUrl: postcallSurveyUrl } }
+      };
+    }
+  } catch (e) {
+    console.error('Unable to parse post call values from environment variables');
```

Comment:
I think throwing here makes sense. The app cannot start because the postCall config is invalid.

Reply:
But postcall is optional.. so if it is undefined/invalid throwing makes sense yes, but the app would still start and the rest would work BAU i think.

Reply:
Great point